apps/backend-sls/test/integration.test.ts

```diff
@@ -1,13 +1,241 @@
+        }
+      `,
+      })
+
+    expect(response.headers['set-cookie']).toBeDefined()
+    userCookie = response.headers['set-cookie'][0]
+
+    expect(response.body).toMatchInlineSnapshot(`
+      Object {
+        "data": Object {
+          "loginUser": "6f45065c-447f-4259-818c-c6f6b477eb48",
+        },
+        "extensions": Object {
+          "responseCache": Object {
+            "invalidatedEntities": Array [],
+          },
+        },
+      }
+    `)
+  })
+
+  it('allows the user to create a session', async () => {
+    const response = await request(app)
+      .post('/api/graphql')
+      .set('Cookie', [userCookie])
+      .send({
+        query: `
+        mutation {
+            createSession(name: "Test Session", blocks: [{ questionIds: [0, 1] }, { questionIds: [2] }]) {
+                id
+                status
+                activeBlock
+                blocks {
+                    id
+                    status
+                }
+            }
+        }
+      `,
+      })
+
+    expect(response.body).toMatchInlineSnapshot(`
+      Object {
+        "data": Object {
+          "createSession": Object {
+            "activeBlock": -1,
+            "blocks": Array [
+              Object {
+                "id": 4,
+                "status": "SCHEDULED",
+              },
+              Object {
+                "id": 5,
+                "status": "SCHEDULED",
+              },
+            ],
+            "id": "b9fcdfad-f58d-4a44-8dae-a2a5dfe778f8",
+            "status": "PREPARED",
+          },
+        },
+        "extensions": Object {
+          "responseCache": Object {
+            "invalidatedEntities": Array [
+              Object {
+                "id": "b9fcdfad-f58d-4a44-8dae-a2a5dfe778f8",
+                "typename": "Session",
+              },
+              Object {
+                "id": 4,
+                "typename": "SessionBlock",
+              },
+              Object {
+                "id": 5,
+                "typename": "SessionBlock",
+              },
+            ],
+          },
+        },
+      }
+    `)
+
+    session = response.body.data.createSession
+  })
+
+  it('allows the user to start a session', async () => {
+    const response = await request(app)
+      .post('/api/graphql')
+      .set('Cookie', [userCookie])
+      .send({
+        query: `
+        mutation {
+            startSession(id: "${session.id}") {
+                id
+                status
+                activeBlock
+                blocks {
+                    id
+                    status
+                }
+            }
+        }
+      `,
+      })
+
+    expect(response.body).toMatchInlineSnapshot(`
+      Object {
+        "data": Object {
+          "startSession": Object {
+            "activeBlock": -1,
+            "blocks": null,
+            "id": "b9fcdfad-f58d-4a44-8dae-a2a5dfe778f8",
+            "status": "RUNNING",
+          },
+        },
+        "extensions": Object {
```

Comment:
The cache for the session that we queried above was invalidated, because we directly mutated the session and returned the changes (what I meant by heuristics above).